stdlib/map.mli

```diff
@@ -223,7 +223,25 @@ module type S =
     (** Same as {!Map.S.map}, but the function receives as arguments both the
        key and the associated value for each binding of the map. *)
 
+    (** {6 Iterators} *)
 
+    type 'a cursor
+    (** Cursor to iterate over bindings
+        @since NEXT_RELEASE *)
+
+    val cursor_start : 'a t -> 'a cursor
+    (** Iterate on values, in ascending order of their corresponding key
+        @since NEXT_RELEASE *)
+
+    val cursor_start_at : key -> 'a t -> 'a cursor
+    (** [cursor_start_at k m] iterates on a subset of the bindings of [m],
+        in ascending order, from key [k] or above.
+        @since NEXT_RELEASE *)
+
+    val cursor_next : 'a cursor -> (key * 'a * 'a cursor) option
+    (** [cursor_next c] returns the next element and a new cursor
```

Comment:
I would use `is` rather than `returns`, describe the value not the fact that a function is being called.